### PR TITLE
Refactoring of the bolsh-arc64-event-logger branch

### PR DESCRIPTION
**Pull request by pavelvkozlov**
This PR is a result of refactoring of the [bolsh-arc64-event-logger](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/tree/bolsh-arc64-event-logger) branch. Here I've collected most of commits from the bolsh-arc64-event-logger branch. 

**What this branch do:**
The main idea of this branch is providing most valuable changes from the [bolsh-arc64-event-logger](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/tree/bolsh-arc64-event-logger) branch to the ```arc64``` branch. Some of commits in this branch have been modified (removed changes of clock freq and base address). But main code hasn't changed. 
- This branch provides change of the core clock frequency to 20MHz.

**What this branch doesn't do:**
I've dropped commits:
- [ARC: Low level Event logger for ARCv3](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/b7990355ad00df678ba6ced0e8bf605faf83e308)
- [ARCv2: Add PMU SLC support](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/260eccf326ca5de3d12411b29feb6ec92bfdaed8)
- [ARCv3: Add cluster PCTs driver](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/49cda94869225b927daa5a4241d488c899e37d09)

As they deserve their own branches, where they can be developed and prepared for merge into the ```arc64```
Also dropped commits:
- [ARC: Added check whether there is a L2$ in the system for HWPTW setup.](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/24fcaa1ada9fcc73f833619d86343cd74b066b50) (linked with event loger)
- [Change default clock frequency for core clock and UART clock to 30MHz.](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/ef67401a9bb3ad47606748257ffa5747bac645a9)

Also:
- This branch doesn't change base address to 0 for the HS5x, it will be done in separate branch. 

I have some doubts about commits:
[ARCv3: increase mmap address space for HS6x](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/0a21bd59e5afb1567f9f2482bd25eab8c177664a)
[ARCv3: Add the ability to change the size of mapped kernel address space](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/f41411a885ffbeed9765f2b68ec217b77d4f3931)

I think some parameters of System Memory Map can be changed, but as current variant was widly used and tested, I think I should offer my changes in a separate branch after merge of these changes.

Compatible with current buildroot. Tested on QEMU and nSIM arc32/arc64.

### CODE REVIEW

**Review by xxkent — APPROVED** [review]

File: arch/arc/include/asm/processor.h [hunk 1]
```diff
@@ -152,6 +152,6 @@ extern unsigned long __get_wchan(struct task_struct *p);
 /* This decides where the kernel will search for a free chunk of vm
  * space during mmap's.
  */
-#define TASK_UNMAPPED_BASE      (TASK_SIZE / 3)
+#define TASK_UNMAPPED_BASE      0x10000000
```
Comment (xxkent) [inline arch/arc/include/asm/processor.h]: Why do we use exactly 0x10000000, 256Mb? Let's try to use more universal approach using TASK_SIZE define.

**Comment by pavelvkozlov** [comment]
Dropped commit:  [ARCv3: Add option to select chunk size of kernel mapping.](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/65b4c6b5bf16de765e0278cc552b8bb7f9c88917)
This commit prevents kernel boot and causes to CI fails in configuration with 64K pages. 
It is open question should we add this commit to the ```arc64``` branch.  This commit offers tradeof of some benefits from using large blocks (2Mb PMD) and increasing the kernel image size. It doesn't work in all configurations and perhaps extra option should be added to the menuconfig to enable this feature. 
It should be dropped or reworked and fixed in separate branch.

**Comment by pavelvkozlov** [comment]
Fixed build of arc64 configuration with 16K pages after drop of commit: [ARCv3: Add option to select chunk size of kernel mapping.](https://github.com/foss-for-synopsys-dwc-arc-processors/linux/commit/65b4c6b5bf16de765e0278cc552b8bb7f9c88917)

**Review by xxkent — APPROVED** [review]

File: arch/arc/include/asm/processor.h (hunk 1 above)
Comment (pavelvkozlov) [reply arch/arc/include/asm/processor.h]: I agree, this change is unneeded, let's move vm for mmap back (TASK_SIZE / 3) is a good value.

**Pull request merged**